crates/flowctl/src/raw/discover.rs

```diff
@@ -0,0 +1,245 @@
+
+        // Create a catalog with the discovered bindings
+        for binding in bindings.iter() {
+            let collection_name = format!("{tenant}/{}", binding.recommended_name);
+            let schema_file = format!("collections/{}.schema.yaml", binding.recommended_name);
+            let collection = Collection::new(collection_name);
+            std::fs::write(&format!("{root}/{schema_file}"), raw_json_to_yaml(&binding.document_schema_json)?)?;
+
+            capture_bindings.push(
+                CaptureBinding {
+                    target: collection.clone(),
+                    resource: RawValue::from_string(binding.resource_config_json.clone())?.into(),
+                }
+            );
+
+            collections.insert(collection,
+                CollectionDef {
+                    schema: Some(Schema::new(serde_json::from_value(json!(schema_file))?)),
+                    write_schema: None,
+                    read_schema: None,
+                    key: CompositeKey::new(binding.key.iter().map(JsonPointer::new).collect::<Vec<JsonPointer>>()),
+                    derive: None,
+                    derivation: None,
+                    projections: Default::default(),
+                    journals: Default::default(),
+                }
+            );
+        };
+
+        let capture_name = format!("{tenant}/{connector_name}");
+        let catalog = Catalog {
+            captures: BTreeMap::from([
+                (Capture::new(capture_name),
+                CaptureDef {
+                    endpoint: CaptureEndpoint::Connector(ConnectorConfig {
+                        image: image.to_string(),
+                        config: serde_json::value::to_raw_value(&config_file)?.into(),
+                    }),
+                    bindings: capture_bindings,
+                    interval: CaptureDef::default_interval(),
+                    shards: ShardTemplate::default(),
+                })
+            ]),
+            collections,
+            ..Default::default()
+        };
+
+        std::fs::write(&format!("{root}/{catalog_file}"), to_yaml(&catalog)?)?;
+    } else {
+        // Otherwise send a Spec RPC and use that to write a sample config file
+        let spec_output = docker_run(image, Request {
+            spec: Some(request::Spec {
+                connector_type: ConnectorType::Image.into(),
+                config_json: "{}".to_string(),
+            }),
+            ..Default::default()
+        }).await?;
+
+        let config_schema_json = serde_json::from_str::<serde_json::Value>(&spec_output.spec.unwrap().config_schema_json)?;
+
+        let config = jsonschema_to_sample_json(&config_schema_json, "")?;
+
+        std::fs::write(&format!("{root}/{config_file}"), to_yaml(&config)?)?;
+    }
+
+
+    Ok(())
+}
+
+fn jsonschema_to_sample_json(schema: &serde_json::Value, root_ptr: &str) -> Result<serde_json::Map<String, serde_json::Value>, anyhow::Error> {
```

Comment:
This is duplicating what `doc::inference` will give you from the schema -- use that. It will also be less brittle.

[Here's a link to the implementation](https://github.com/estuary/flow/blob/ed4c28da4296e7bedd370b307951641e56ac9b20/go/flowctl-go/cmd-discover.go#L257-L279) we used to have, that will generate a YAML with comments, defaults, etc.